src/scipp/logging.py

```diff
@@ -0,0 +1,183 @@
+        logging.Formatter('[%(asctime)s] %(levelname)-8s : %(message)s',
+                          datefmt='%Y-%m-%dT%H:%M:%S'))
+    return handler
+
+
+if running_in_jupyter():
+    from ipywidgets import HTML
+
+    class LogWidget(HTML):
+        """
+        Widget that displays log messages in a table.
+        """
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+            self._rows_str = ''
+
+        def add_message(self, name: str, time_stamp: str, level: str,
+                        message: str) -> None:
+            """
+            Add a message to the output.
+            :param name: Name of the logger.
+            :param time_stamp: Formatted time when the message was recorded.
+            :param level: Name of the log level.
+            :param message: The actual message.
+            """
+            self._rows_str += self._format_row(name, time_stamp, level, message)
+            self._update()
+
+        @staticmethod
+        def _format_row(name: str, time_stamp: str, level: str, message: str) -> str:
+            return (f'<tr class="sc-log-{level.lower()}">'
+                    f'<td class="sc-log-time-stamp">[{html.escape(time_stamp)}]</td>'
+                    f'<td class="sc-log-level">{level}</td>'
+                    f'<td class="sc-log-message">{html.escape(message)}</td>'
+                    f'<td class="sc-log-name">&lt;{html.escape(name)}&gt;</td>'
+                    f'</tr>')
+
+        def _update(self) -> None:
+            self.value = f'<div class="sc-log"><table>{self._rows_str}</table></div>'
+
+        def clear(self) -> None:
+            """
+            Remove all output from the widget.
+            """
+            self._rows_str = ''
+            self._update()
+
+
+def make_log_widget() -> 'LogWidget':
+    """
+    Create and return a new ``LogWidget`` for use with ``WidgetHandler``
+    in a Jupyter notebook.
+
+    :raises RuntimeError: If Python is not running in Jupyter.
+    """
+    if not running_in_jupyter():
+        raise RuntimeError('Cannot construct a logging widget because '
+                           'Python is not running in a Jupyter notebook.')
+    else:
+        return LogWidget()
+
+
+def get_log_widget() -> Optional['LogWidget']:
+    """
+    Return the log widget used by the global scipp logger.
+    If multiple widget handlers are installed, only the first one is returned.
+    If no widget handler is installed, returns ``None``.
+    """
+    handler = get_widget_handler()
+    if handler is None:
+        return None
+    return handler.widget
+
+
+def display_logs() -> None:
+    """
+    Display the log widget associated with the global scipp logger.
+    """
+    widget = get_log_widget()
+    if widget is None:
+        raise RuntimeError(
+            'Cannot display log widgets because no widget handler is installed.')
+
+    from IPython.display import display
+    inject_style()
+    display(widget)
```

Comment:
It is a bit odd that the log window is empty initial (hard to see and detach view). Maybe we could add a startup message stating some subset/combination of scipp version / licence / docs links / citation?

Reply:
Not sure this is a good idea. Imagine numpy did that and log a message in every package / application that uses it. We can provide a function that logs some basic information if users want to put that into  their logs. (Could be useful to have the version.)

I can just add a header to the widget and make it visible this way.

Reply:
I am not suggesting to log this when importing scipp, rather when this specific widget is created? Or would you consider that too annoying as well?

Reply:
No, this could work.
But have a look at the current version. The widget shows up at its full size from the start.